disable.ps1

```diff
@@ -148,150 +69,388 @@ function Resolve-ZenyaErrorMessage {
+                    Method          = 'PATCH'
+                    Body            = ([System.Text.Encoding]::UTF8.GetBytes($body))
+                    ContentType     = "application/json;charset=utf-8"
+                    UseBasicParsing = $true
+                }
+
+                if (-Not($actionContext.DryRun -eq $true)) {
+                    Write-Verbose "Updating account [$($currentAccount.Username)]. AccountReference: $($actionContext.References.Account | ConvertTo-Json -Depth 10). Body: $($updateAccount | ConvertTo-Json -Depth 10)"
+        
+                    $updatedAccount = Invoke-RestMethod @splatWebRequest -Verbose:$false
+                    # Set the correct account reference
+                    $outputContext.AccountReference = [PSCustomObject]@{
+                        Id       = $updatedAccount.id
+                        Username = $updatedAccount.userName
+                    }
+        
+                    $outputContext.AuditLogs.Add([PSCustomObject]@{
+                            # Action  = "" # Optional
+                            Message = "Successfully updated account [$($currentAccount.Username)]. AccountReference: $($outputContext.AccountReference | ConvertTo-Json -Depth 10). Old values: $($changedPropertiesObject.oldValues | ConvertTo-Json -Depth 10). New values: $($changedPropertiesObject.newValues | ConvertTo-Json -Depth 10)"
+                            IsError = $false
+                        })
+                }
+                else {
+                    Write-Warning "DryRun: Would update account [$($currentAccount.Username)]. AccountReference: $($actionContext.References.Account | ConvertTo-Json -Depth 10). Old values: $($changedPropertiesObject.oldValues | ConvertTo-Json -Depth 10). New values: $($changedPropertiesObject.newValues | ConvertTo-Json -Depth 10)"
+                }
+                break
+            }
+            catch {
+                $ex = $PSItem
+                $errorMessage = Get-ErrorMessage -ErrorObject $ex
+        
+                Write-Verbose "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
+                Write-Verbose "URI: $($splatWebRequest.Uri)"
+                Write-Verbose "Body: $($body)"
+        
+                $outputContext.AuditLogs.Add([PSCustomObject]@{
+                        # Action  = "" # Optional
+                        Message = "Error updating account [$($currentAccount.Username)]. AccountReference: $($actionContext.References.Account | ConvertTo-Json -Depth 10). Error Message: $($errorMessage.AuditErrorMessage)"
+                        IsError = $true
+                    })
+        
+                # Throw terminal error
+                throw
+            }
+        }
+        'NoChanges' {
+            if (-not($dryRun -eq $true)) {
+                Write-Verbose "Skipped updating account [$($account.Username)]. Reason: No changes"
```

Comment:
If there are no changes, do we still log the disable action?

Reply:
Only as dryrun or verbose logging, not as auditlog.